### PR TITLE
API of the timer coordinator module

### PR DESCRIPTION
**Pull request by hubertmis**

### CODE REVIEW

File: src/nrf_802154_timer_coord.h [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ *
+ * @param[in]  event_addr  Address of given event.
+ */
+void nrf_802154_timer_coord_timestamp_prepare(uint32_t event_addr);
+
+/**
+ * @brief Get timestamp of the recently prepared event.
+ *
+ * If recently prepared event occurred a few times since preparation, this function returns
+ * timestamp of the first occurrence.
```
[the remark this answers is not in the document]
Comment (hubertmis) [reply src/nrf_802154_timer_coord.h]: I can imagine that we pass END event here in radio RX state. END event would be triggered when frame is received and when ACK is transmitted. We want to get timestamp of first occurrence.

File: src/nrf_802154_timer_coord.h [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+void nrf_802154_timer_coord_init(void);
+
+/**
+ * @brief Uninitialize the Timer Coordinator module.
+ *
+ */
+void nrf_802154_timer_coord_uninit(void);
+
+/**
+ * @brief Start timer coordinator.
```
Comment (LuDuda) [inline src/nrf_802154_timer_coord.h]: nit: align to above "the Timer Coordinator"

File: src/nrf_802154_timer_coord.h [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ * 
+ * This function starts the HP timer and synchronizes it to the LP timer.
```
Comment (LuDuda) [inline src/nrf_802154_timer_coord.h]: nit: synchronize with?

File: src/nrf_802154_timer_coord.h [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+/**
+ * @brief Stop timer coordinator.
+ *
+ * This function stops the HP timer.
+ */
+void nrf_802154_timer_coord_stop(void);
+
+/**
+ * @brief Prepare getting precise timestamp of given event.
```
Comment (LuDuda) [inline src/nrf_802154_timer_coord.h]: Is it address of hardware event e.g. EVENT_END? I guess so, then i would specify this in the dox.

File: src/nrf_802154_timer_coord.h [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+/**
+ *@}
+ **/
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* NRF_802154_RSCH_H_ */
```
Comment (LuDuda) [inline src/nrf_802154_timer_coord.h]: NRF_802154_TIMER_COORD_H_

**Comment by hubertmis** [comment]
@LuDuda I don't want to name this module `timestamp sth`, because it is intended to do much more than just timestamps. It coordinates HP timer with LP timer (that's where current name proposal came from :) ). The first feature of HP timer we are implementing is timestamp indeed. But hopefully we will implement more features and renaming would be a waste.

**Pull request merged**